src/history/html5.js

```diff
@@ -72,7 +72,7 @@ export class HTML5History extends History {
 }
 
 export function getLocation (base: string): string {
-  let path = window.location.pathname
```

Comment:
I think we should use `decodeURI` to keep separators (such as `/` and `&`) encoded

Reply:
@posva changed